source/code/source/playfab/PlayFabEventPipeline.cpp

```diff
@@ -23,33 +23,54 @@ namespace PlayFab
         maximalNumberOfBatchesInFlight(defaultMaxBatchesInFlight),
         readBufferWaitTime(defaultReadBufferWaitTimeInMs),
         authenticationContext(nullptr),
-        emitType(PlayFabEventPipelineType::PlayFabPlayStream)
+        emitType(PlayFabEventPipelineType::PlayFabPlayStream),
+        useBackgroundThread(true)
     {
     }
 
-    PlayFabEventPipelineSettings::PlayFabEventPipelineSettings(PlayFabEventPipelineType type) :
+    PlayFabEventPipelineSettings::PlayFabEventPipelineSettings(PlayFabEventPipelineType type, bool useBackgroundThread) :
         bufferSize(defaultBufferSize),
         maximalNumberOfItemsInBatch(defaultMaxItemsInBatch),
         maximalBatchWaitTime(defaultMaxBatchWaitTimeInSeconds),
         maximalNumberOfBatchesInFlight(defaultMaxBatchesInFlight),
         readBufferWaitTime(defaultReadBufferWaitTimeInMs),
         authenticationContext(nullptr),
-        emitType(type)
+        emitType(type),
+        useBackgroundThread(useBackgroundThread)
     {
     }
 
     PlayFabEventPipeline::PlayFabEventPipeline(const std::shared_ptr<PlayFabEventPipelineSettings>& settings) :
+        batchCounter(0),
         buffer(settings->bufferSize),
         isWorkerThreadRunning(false)
     {
         eventsApi = std::make_shared<PlayFabEventsInstanceAPI>(PlayFabSettings::staticPlayer);
 
         this->settings = settings;
         this->batchesInFlight.reserve(this->settings->maximalNumberOfBatchesInFlight);
-        this->Start();
+        if (this->settings->useBackgroundThread)
+        {
+            this->Start();
+        }
     }
 
     PlayFabEventPipeline::~PlayFabEventPipeline()
+    {
+        Stop();
+    }
+
+    void PlayFabEventPipeline::Start()
+    {
+        // start worker thread
+        this->isWorkerThreadRunning = true;
```

Comment:
This function should check if isWorkerThreadRunning is already true and do error handling.